### ISSUE TITLE
[Bug Report] Cohere compatibility mode double-applies logit_scale in forward

### ISSUE THREAD

**Issue by emerardd** [TransformerLensOrg/TransformerLens#1726]
**Describe the bug**

For Cohere and Cohere2 bridges, weight processing folds `config.logit_scale` into the live
unembedding weight even though the wrapped Hugging Face forward still multiplies the resulting logits
by the same scale. The first `process_weights()` / compatibility-mode call therefore applies the scale
twice relative to the raw Bridge output.

With `logit_scale=0.5`, processing changes the output to exactly `raw * 0.5`. The call returns
normally, so this is a silent numerical correctness problem for logits and weight-space analyses.

`Cohere2ArchitectureAdapter` inherits the same preprocessing behavior from
`CohereArchitectureAdapter`.

**Code example**

This reproduction uses a randomly initialized tiny model from config and does not access the Hub:

```python
import torch
from transformers import CohereConfig, CohereForCausalLM

from transformer_lens.model_bridge.sources import build_bridge_from_module

torch.manual_seed(0)
config = CohereConfig(
    vocab_size=32,
    hidden_size=16,
    intermediate_size=32,
    num_hidden_layers=1,
    num_attention_heads=4,
    num_key_value_heads=2,
    max_position_embeddings=32,
    logit_scale=0.5,
    attention_bias=False,
    use_qk_norm=False,
    tie_word_embeddings=True,
    bos_token_id=1,
    eos_token_id=2,
    pad_token_id=0,
)
hf_model = CohereForCausalLM(config).eval()
bridge = build_bridge_from_module(
    hf_model,
    "CohereForCausalLM",
    hf_config=config,
    dtype=torch.float32,
    device="cpu",
)

tokens = torch.tensor([[1, 2, 3, 4]])
raw = bridge(tokens).detach()

bridge.enable_compatibility_mode(
    disable_warnings=True,
    fold_ln=False,
    center_writing_weights=False,
    center_unembed=False,
    fold_value_biases=False,
)
processed = bridge(tokens).detach()

print((processed - raw * 0.5).abs().max().item())
```

Current output on `dev` at `572946b7`:

```text
0.0
```

The cause is the interaction between:

- `CohereArchitectureAdapter.preprocess_weights()`, which multiplies `unembed.weight` and
  `unembed.bias` by `cfg.logit_scale`; and
- `CohereForCausalLM.forward()`, which still multiplies the logits produced from that live weight by
  its instance-level `logit_scale`.

**Expected behavior**

- The first compatibility-mode weight processing call should preserve Cohere's single
  `logit_scale` application in forward.
- The scale should remain folded into the unembedding weight for compatibility-mode weight-space
  analyses.
- Tied input embeddings must remain unscaled.
- `cfg.logit_scale` should retain the architecture metadata, and `logit_scale=1.0` should remain a
  no-op.

**Suggested scope**

Keep the fix focused on the Cohere/Cohere2 output-scale contract and add a network-free tiny-config
regression covering both direct `process_weights()` and the first compatibility-mode call. Existing
raw-HF parity, tied-embedding, and Jacobian-lens scale behavior should remain intact.

**System Info**

- Installed from source on the current `dev` branch (`572946b7`)
- Windows 11, CPU
- Python 3.12.10
- PyTorch 2.10.0+cpu
- Transformers 5.13.0

**Additional context**

Compatibility mode is documented as a one-shot transformation because generic centering operations
are not reversible or generally idempotent. A second call producing `raw * 0.25` helped isolate the
compounding fold, but this report does not propose making the whole compatibility-mode pipeline
repeatable.

Open and closed Issue/PR searches were run for `Cohere compatibility logit_scale` and the broader
`logit_scale` term. Closed #1325 and merged #1348 cover non-trivial-scale test parameterization, not
the first-call forward double application.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

**Comment by emerardd** [comment]
I'll work on a focused fix for the first-call forward double application, with network-free tiny-config regressions for direct process_weights() and compatibility mode. I'll keep the scope limited to the Cohere/Cohere2 output-scale contract.
